### ISSUE TITLE
"Can't toast on a thread that has not called Looper.prepare()"

### ISSUE THREAD

**Issue by lalalasupa0** [aniyomiorg/aniyomi#1835]
### Steps to reproduce

Open extension (any, really. I sed hanime.tv and some random entry)
Download ep with external downloader (I'm using 1DM+)

### Expected behavior

![Screenshot_20241130-191045](https://github.com/user-attachments/assets/55eba2c1-34c6-4fd3-bf81-8cd37ecc494c)

### Actual behavior

![Screenshot_20241130-191059](https://github.com/user-attachments/assets/55303821-4d16-4b78-9c81-a44d6f1303c3)

And notification
`Can't toast on a thread that has not called Looper.prepare()`

### Aniyomi version

App version: 0.16.4.3-8019 (standard, 863720712, 126, 2024-11-09T21:51:27Z) Android version: 11 (SDK 30; build RPNS31.Q1-51-40-16-12) Android build ID: RPNS31.Q1-51-40-16-12 Device brand: motorola Device manufacturer: motorola Device name: nairo (nairo_a) Device model: motorola one 5G WebView: Android System WebView 130.0.6723.107 MPV version: 4d32db2 Libplacebo version: v7.349.0 FFmpeg version: n7.0

### Android version

Android 11

### Device

Motorola One 5G

### Other details

Idk if it happens with any other downloader

It didn't happen before with the same anime and episodes, I think it came with last update

I does create the folder in /downloads/<anime>/ but the episode folder name comes with a "_tmp"
Only after manually renaming them is when it appears in aniyomi like the screenshot in *Expected behavior*

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the extension's repository.
- [X] I have gone through the [FAQ](https://aniyomi.org/docs/faq/general) and [troubleshooting guide](https://aniyomi.org/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.4.3](https://github.com/aniyomiorg/aniyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by juniornff** [comment]
Same with Android 12 on a Poco X3 Pro using [ADM](https://play.google.com/store/apps/details?id=com.dv.adm)

**Comment by AngelKrak** [comment]
Same issue with Android 12 using 1DM+

App version: 0.16.4.3-8065 (standard, 25dd3ea69, 128, 2025-04-01T09:26:25Z)
Android version: 12 (SDK 31; build HD1910_11_F.22)
Android build ID: HD1910_11_F.22
Device brand: OnePlus
Device manufacturer: OnePlus
Device name: OnePlus7TPro (OnePlus7TPro)
Device model: HD1910
WebView: Android System WebView 135.0.7049.100
MPV version: 6764488
Libplacebo version: v7.349.0
FFmpeg version: n7.1
